shell/error_explainer.go

```diff
@@ -0,0 +1,45 @@
+package shell
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/hashicorp/go-multierror"
+)
+
+// terraformErrorsMatcher List of errors that we know how to explain to the user. The key is a regex that matches the error message, and the value is the explanation.
+var terraformErrorsMatcher = map[string]string{
+	"(?s).*Error refreshing state: AccessDenied: Access Denied(?s).*": "You don't have access to the S3 bucket where the state is stored. Check your AWS credentials and permissions.",
```

Comment:
NIT / just curious: is there any way we could output the actual bucket name that they were not able to access? That might make it even easier to understand what's happening?

Reply:
Having bucket name may help in troubleshooting, but since name is not included in output it makes it more complicated to be printed.

Additionally, it will require implementing some kind of templating language on explanations to allow inserting custom values.